### DISCUSSION TITLE
How do I configure firewall for GitHub Container Registry

### DISCUSSION THREAD

**Discussion by lpc921** [community/community#118629] [Packages] [answered]
### Select Topic Area

Question

### Body

When running docker pull from GitHub Container Registry, I got the following message:
`Download failed, retrying (2/5): dial tcp 185.199.111.154:443: i/o timeout`

We have allowed `ghcr.io` and `*.github .com` in the firewall.

If I reverse loop up the IP address and run `curl https://cdn-185-199-109-154.github.com`, then the docker pull works

How should the firewall be configured to make docker pull work seamlessly with GHCR? Or maybe there is a docker daemon config to force a DNS lookup?

**Comment by KittyChiu** [comment, marked as answer]
There are more addresses for GitHub Packages.  Use [Meta API](https://docs.github.com/en/rest/meta/meta?apiVersion=2022-11-28) and navigate to `packages` node to get the latest list of addresses for whitelisting.

At the time of typing this comment, the list is
```
        "packages": [
            "mavenregistryv2prod.blob.core.windows.net",
            "npmregistryv2prod.blob.core.windows.net",
            "nugetregistryv2prod.blob.core.windows.net",
            "rubygemsregistryv2prod.blob.core.windows.net",
            "npm.pkg.github.com",
            "npm-proxy.pkg.github.com",
            "npm-beta-proxy.pkg.github.com",
            "npm-beta.pkg.github.com",
            "nuget.pkg.github.com",
            "rubygems.pkg.github.com",
            "maven.pkg.github.com",
            "docker.pkg.github.com",
            "docker-proxy.pkg.github.com",
            "containers.pkg.github.com",
            "*.github.com",
            "*.pkg.github.com",
            "*.ghcr.io",
            "*.githubassets.com",
            "*.githubusercontent.com"
        ],
```

**Reply by lpc921** [reply]
Thanks for your answer. I needed to whitelist `pkg-containers.githubusercontent.com`

**Reply by grandmas-favorite** [reply]
Nice one @KittyChiu!

**Reply by hassanbaloshi** [reply]
> Thanks for your answer. I needed to whitelist `pkg-containers.githubusercontent.com`

I want logain ip idrees

**Reply by AbhishekPratap9** [reply]
https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-rubygems-registry

If having tough time to find the packages, check out here. And please confirm also

**Comment by hassanbaloshi** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> When running docker pull from GitHub Container Registry, I got the following message: `Download failed, retrying (2/5): dial tcp 185.199.111.154:443: i/o timeout`
> 
> We have allowed `ghcr.io` and `*.github .com` in the firewall.
> 
> If I reverse loop up the IP address and run `curl https://cdn-185-199-109-154.github.com`, then the docker pull works
> 
> How should the firewall be configured to make docker pull work seamlessly with GHCR? Or maybe there is a docker daemon config to force a DNS lookup?

> __

**Comment by hassanbaloshi** [comment]
![Screenshot_20250506_153019](https://github.com/user-attachments/assets/ec050d88-f976-4ec9-8878-7f945e63e696)

**Comment by hassanbaloshi** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> When running docker pull from GitHub Container Registry, I got the following message: `Download failed, retrying (2/5): dial tcp 185.199.111.154:443: i/o timeout`
> 
> We have allowed `ghcr.io` and `*.github .com` in the firewall.
> 
> If I reverse loop up the IP address and run `curl https://cdn-185-199-109-154.github.com`, then the docker pull works
> 
> How should the firewall be configured to make docker pull work seamlessly with GHCR? Or maybe there is a docker daemon config to force a DNS lookup?


U

**Comment by hassanbaloshi** [comment]
192.168.207.153

**Reply by hassanbaloshi** [reply]
> 192.168.207.153

**Comment by aizaz8999-wq** [comment]
You can contact me
